File: code_samples/lua/functional_monads.lua
-- Functional: Monads
-- AI/ML Training Sample

Monads = {}
Monads.__ index = Monads

function Monads:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Monads:process(input)
    self.data = input
end

function Monads:getData()
    return self.data
end

function Monads:validate()
    return #self.data > 0
end

-- Example usage
local instance = Monads:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/rust/functional_monads.rs
//! Functional: Monads
//! AI/ML Training Sample

pub struct Monads {
    data: String,
}

impl Monads {
    pub fn new() -> Self {
        Self {
            data: String::new(),
        }
    }
    
    pub fn process(&mut self, input: &str) {
        self.data = input.to_string();
    }
    
    pub fn get_data(&self) -> &str {
        &self.data
    }
    
    pub fn validate(&self) -> bool {
        !self.data.is_empty()
    }
}

fn main() {
    let mut instance = Monads::new();
    instance.process("example");
    println!("Data: {}", instance.get_data());
    println!("Valid: {}", instance.validate());
}
